primitives/parachain-inherent/src/mock.rs

```diff
@@ -59,6 +61,30 @@ pub struct MockValidationDataInherentDataProvider {
 	pub raw_horizontal_messages: Vec<(ParaId, Vec<u8>)>,
 }
 
+pub trait GenerateRandomness<I> {
+	fn generate_randomness(&self, input: I) -> relay_chain::Hash;
+}
+
+impl GenerateRandomness<u64> for () {
+	/// Default implementation uses relay epoch as randomness value
+	/// A more seemingly random implementation may hash the relay epoch instead
+	fn generate_randomness(&self, input: u64) -> relay_chain::Hash {
+		let mut mock_randomness: [u8; 32] = [0u8; 32];
+		mock_randomness[..8].copy_from_slice(&input.to_be_bytes());
+		mock_randomness.into()
+	}
+}
+
+/// Parameters for how the Mock inherent data provider should inject the relay
+/// epoch index and BABE one epoch ago randomness.
+pub struct MockRandomnessConfig<R> {
+	/// Number of parachain blocks per relay chain epoch
+	/// Mock epoch is computed by dividing `current_para_block` by this value.
+	pub para_blocks_per_epoch: u32,
```

Comment:
Why is that part of the `RandomnessConfig`?

Reply:
The default configuration updates the randomness whenever the epoch changes so I put them together, but now I'm rethinking it.

Reply:
I'll move the fields into `MockValidationDataInherentDataProvider` and get rid of this struct as that is preferred.